packages/guess-parser/src/angular/index.ts

```diff
@@ -194,13 +194,14 @@ const readChildren = (
   return null;
 };
 
-const getModulePathFromRoute = (parentPath: string, loadChildren: string) => {
-  const childModule = loadChildren.split('#')[0] + '.ts';
+const getModulePathFromRoute = (parentPath: string, loadChildren: string, program: ts.Program, host: ts.CompilerHost) => {
+  const childModule = loadChildren.split('#')[0];
+  const childModuleFile = childModule + '.ts';
   if (loadChildren.startsWith('.')) {
-    return join(dirname(parentPath), childModule);
+    return join(dirname(parentPath), childModuleFile);
   }
   const parentSegments = dirname(parentPath).split(sep);
```

Comment:
I was hoping to get rid of this but `resolveModuleName` seems to get confused when the route is relative to the app root, which in Angular is possible.

Reply:
I think this is because TypeScript does not even attempt to look up a file if it does not start with a dot. I guess it does try to look up the first part of those paths as a node module instead. So if we define `loadChildren: 'app/qux/qux.module'` TypeScript will probably look for a node module named 'app' and if it exists it would attempt to find the 'qux/qux.module' file in it.

Anyway I wondered if we could help TypeScript by prepending a './' to tell it that we want it to look for a file and not a module. I did a quick test but it doesn't work right now. I think it's because the test fixture is inconsistent when it comes to the the old `loadChildren` syntax. The path defined [here](https://github.com/guess-js/guess/blob/master/packages/guess-parser/test/fixtures/angular/src/app/app-routing.module.ts#L33) would have to be prepended with '../' since the path starts with the parent folder. But [here](https://github.com/guess-js/guess/blob/master/packages/guess-parser/test/fixtures/angular/src/app/foo/foo-routing.module.ts#L18) the path would have to be prepended with './' since it starts with a folder inside the current directory. Did Angular resolve both of them correctly or is that a bug in the test fixture?